godot-core/src/builtin/packed_array.rs

```diff
@@ -258,6 +258,34 @@ macro_rules! impl_packed_array {
                 self.as_inner().sort();
             }
 
+            /// Extracts a slice of the entire array.
+            pub fn as_slice<'a>(&'a self) -> &'a [$Element] {
+                let len = self.len();
+
+                if len == 0 {
+                    return &[];
+                }
+
+                let data = self.ptr(0);
+                unsafe {
+                    std::slice::from_raw_parts(data, len)
```

Comment:
You should probably add a `// SAFETY:` comment here and in the other function explaining why this is safe to do.